File: framework/src/gmsec4/c/field/boolean_field.cpp
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */



/** @file boolean_field.cpp
 *
 *  @brief This file contains functions for the management of a BooleanField object.
 */


#include <gmsec4/c/field/boolean_field.h>

#include <gmsec4/field/BooleanField.h>

#include <gmsec4/Exception.h>
#include <gmsec4/Status.h>


using namespace gmsec::api;


GMSEC_Field CALL_TYPE booleanFieldCreate(const char* fieldName, GMSEC_BOOL value, GMSEC_Status status)
{
	BooleanField* field = NULL;
	Status        result;

	try
	{
		field = new BooleanField(fieldName, value == GMSEC_TRUE);
	}
	catch (Exception& e)
	{
		result = Status(e);
	}

	if (status)
	{
		*(reinterpret_cast<Status*>(status)) = result;
	}

	return field;
}


GMSEC_BOOL CALL_TYPE booleanFieldGetValue(const GMSEC_Field field, GMSEC_Status status)
{
	const Field*        f         = reinterpret_cast<const Field*>(field);
	const BooleanField* boolField = dynamic_cast<const BooleanField*>(f);
	GMSEC_BOOL          value     = GMSEC_FALSE;
	Status              result;

	if (boolField)
	{
		value = boolField->getValue() ? GMSEC_TRUE : GMSEC_FALSE;
	}
	else
	{
		result = Status(FIELD_ERROR, INVALID_FIELD, "Field object is not a BooleanField");
	}

	if (status)
	{
		*(reinterpret_cast<Status*>(status)) = result;
	}

	return value;
}

File: nodejs/gmsec5/dox/BooleanField.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file BooleanField.java
 */

package gmsec;


/**
 * Specialized Field class that can be used to retain a boolean value.
 */
public class BooleanField extends Field
{
	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data Boolean data to associate with the field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public BooleanField(name, data);


	/**
	 * Constructor.
	 *
	 * @param name      Name of the field.
	 * @param data      Boolean data to associate with the field.
	 * @param is_header Used to indicate if Field is a header field.
	 *
	 * @throws SWIG_Exception Thrown if the field name is null.
	 * @throws GmsecException Thrown if the field name string is an empty string or is otherwise non-compliant.
	 */
	public BooleanField(name, data, is_header);


	/**
	 * Copy constructor.
	 *
	 * @param other The other BooleanField object to copy.
	 *
	 * @throws SWIG_Exception Thrown if the given BooleanField is null.
	 */
	public BooleanField(other);


	/**
	 * Returns the data stored within the BooleanField.
	 *
	 * @return A Boolean value.
	 */
	public getValue();
}
